controller/kubernetes_pod_controller.go

```diff
@@ -227,6 +233,55 @@ func (kc *KubernetesPodController) handlePodDeletionIfNodeDown(pod *v1.Pod, node
 	return nil
 }
 
+// handlePodDeletionIfVolumeRequestRemount will delete the pod which is using a volume that has requested remount.
+// By deleting the consuming pod, Kubernetes will recreated them, reattaches, and remounts the volume.
+func (kc *KubernetesPodController) handlePodDeletionIfVolumeRequestRemount(pod *v1.Pod) error {
+	// Only delete pod which has controller to make sure that the pod will be recreated by its controller
+	if metav1.GetControllerOf(pod) == nil {
+		return nil
+	}
+
+	volumeList, err := kc.getAssociatedVolumes(pod)
+	if err != nil {
+		return err
+	}
+
+	// Only delete pod which has startTime < vol.Status.RemountRequestAt AND timeNow > vol.Status.RemountRequestAt + delayDuration
+	// The delayDuration is to make sure that we don't repeatedly delete the pod too fast
+	// when vol.Status.RemountRequestAt is updated too quickly by volumeController
+	if pod.Status.StartTime == nil {
+		return nil
+	}
+	podStartTime := pod.Status.StartTime.Time
+	for _, vol := range volumeList {
+		if vol.Status.LastRemountRequestAt == "" {
+			continue
+		}
+		remountRequestAt, err := time.Parse(time.RFC3339, vol.Status.LastRemountRequestAt)
+		if err != nil {
+			return err
+		}
+
+		timeNow := time.Now()
+		delayDuration := time.Duration(int64(5)) * time.Second
+
+		if podStartTime.Before(remountRequestAt) && timeNow.After(remountRequestAt.Add(delayDuration)) {
+			gracePeriod := int64(0)
+			err := kc.kubeClient.CoreV1().Pods(pod.Namespace).Delete(pod.Name, &metav1.DeleteOptions{
```

Comment:
Is it a force deletion(grace period 0)? Should we allow the pod to shutdown properly to release the resources?

Reply:
Thank you! 
Yeah, we should not force delete